tests/Unit/Symfony/OtelSdkBundle/DataCollector/OtelDataCollectorTest.php

```diff
@@ -0,0 +1,165 @@
+        $dataCollector->setExporterData(null);
+        $this->assertEquals(1, count($dataCollector->getData()));
+        $dataCollector->reset();
+        $this->assertEquals(0, count($dataCollector->getData()));
+    }
+
+    public function testGetName(): void
+    {
+        $dataCollector = new OtelDataCollector();
+        $this->assertEquals('otel', $dataCollector->getName());
+    }
+
+    public function testGetClassLocations(): void
+    {
+        $dataCollector = new OtelDataCollector();
+        $this->assertEquals([
+            'class' => 'OtelDataCollectorTest',
+            'file' => __FILE__,
+        ], $dataCollector->getClassLocation(get_class($this)));
+    }
+
+    /**
+     * @dataProvider exporterDataProvider
+     */
+    public function testSetExporterData(?SpanExporterInterface $spanExporter, ?array $expectedExporterData): void
+    {
+        $dataCollector = new OtelDataCollector();
+        $dataCollector->setExporterData($spanExporter);
+        $this->assertEquals($expectedExporterData, $dataCollector->getData()['exporter']);
+    }
+
+    public function testLateCollectWithoutSharedState(): void
+    {
+        $dataCollector = new OtelDataCollector();
+        $dataCollector->lateCollect();
+        $this->assertEquals(['spans' => []], $dataCollector->getData());
+    }
+
+    public function testLateCollectWithTracerSharedState(): void
+    {
+        $sharedState = new TracerSharedState(
+            new RandomIdGenerator(),
+            ResourceInfoFactory::emptyResource(),
+            (new SpanLimitsBuilder())->build(),
+            new AlwaysOnSampler(),
+            [new SimpleSpanProcessor()],
+        );
+
+        $tracer = new Tracer($sharedState, $this->createMock(InstrumentationScopeInterface::class));
+        $dataCollector = new OtelDataCollector();
+        $dataCollector->setTracer($tracer);
+        $dataCollector->lateCollect();
+        $this->assertEquals([], $dataCollector->getData()['spans']);
+        $this->assertEquals([
+            'class' => 'RandomIdGenerator',
+            'file' => (new \ReflectionClass(RandomIdGenerator::class))->getFileName(),
+        ], $dataCollector->getData()['id_generator']);
+        $this->assertEquals([
+            'class' => 'AlwaysOnSampler',
+            'file' => (new \ReflectionClass(AlwaysOnSampler::class))->getFileName(),
+        ], $dataCollector->getData()['sampler']);
+        $this->assertEquals([
+            'class' => 'SimpleSpanProcessor',
+            'file' => (new \ReflectionClass(SimpleSpanProcessor::class))->getFileName(),
+        ], $dataCollector->getData()['span_processor']);
+        $this->assertInstanceOf(Data::class, $dataCollector->getData()['resource_info_attributes']);
+        $this->assertInstanceOf(Data::class, $dataCollector->getData()['span_limits']);
+    }
+
+    public function testLateCollectWithTracerProviderSharedState(): void
+    {
+        $tracerProvider = new TracerProvider(
```

Comment:
As a general observation, the tests don't use mocking which I'd normally expect a unit test to do...

Reply:
I'm trying to use mock but unfortunatly I did'nt success. OtelDataCollector use Reflection in order to use TracerSharedState.
And I did'nt find the right way to create a mock that work with reflection (I get `ReflectionException: Property tracerSharedState does not exist` error).
Do you have some tips for doing this kind of mock ?

Reply:
Sorry, I've never come across that one before. It's not a deal-breaker though, leave it as it is :)